src/catchup/CatchupWork.cpp

```diff
@@ -342,10 +342,12 @@ CatchupWork::getAndMaybeSetHistoryArchiveState()
                     ? CatchupConfiguration::CURRENT
                     : mApp.getHistoryManager().checkpointContainingLedger(
                           mCatchupConfiguration.toLedger());
+
             // Set retries to 10 to ensure we retry enough in case current
             // checkpoint isn't published yet
             mGetHistoryArchiveStateWork = addWork<GetHistoryArchiveStateWork>(
-                toCheckpoint, mArchive, true, 10);
+                toCheckpoint, mArchive, true,
+                mApp.getConfig().HISTORY_ARCHIVE_RETRY_LIMIT);
```

Comment:
Catchup has a few other steps that download files (`DownloadApplyTxsWork`, `DownloadBucketsWork`, etc), but looks like those do not use the new config?